core/persist/PersistenceProvider.ts

```diff
@@ -98,29 +122,10 @@ export abstract class PersistenceProvider<S> {
                 if (rest.getData || rest.setData) type = 'custom';
```

Comment:
I still *wonder* if we would want to take advantage of this change to make the shortcut form above deprecated, and favor a form of {type: 'pref', key: 'MyPreferenceKey'}.  At that point, not really sure if we need pre-registration, if type could have the type of 'string| type of PrefenceProvider'.

Just thinking out loud here to get all options on the table.  Very much like this proposal as written as well

Reply:
Happy to hand this over to you guys to decide - this was my proposal to move things incrementally along and minimize breaking changes.